lightning/src/routing/network_graph.rs

```diff
@@ -1813,4 +1945,184 @@ mod tests {
 		network.write(&mut w).unwrap();
 		assert!(<NetworkGraph>::read(&mut ::std::io::Cursor::new(&w.0)).unwrap() == *network);
 	}
+
+	#[test]
+	fn calling_sync_routing_table() {
+		let (secp_ctx, net_graph_msg_handler) = create_net_graph_msg_handler();
+		let node_privkey_1 = &SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_privkey_1);
+
+		let chain_hash = genesis_block(Network::Testnet).header.block_hash();
+		let first_blocknum = 0;
+		let number_of_blocks = 0xffff_ffff;
+
+		// It should ignore if gossip_queries feature is not enabled
+		{
+			let init_msg = Init { features: InitFeatures::known().clear_gossip_queries() };
+			net_graph_msg_handler.sync_routing_table(&node_id_1, &init_msg);
+			let events = net_graph_msg_handler.get_and_clear_pending_msg_events();
+			assert_eq!(events.len(), 0);
+		}
+
+		// It should send a query_channel_message with the correct information
+		{
+			let init_msg = Init { features: InitFeatures::known() };
+			net_graph_msg_handler.sync_routing_table(&node_id_1, &init_msg);
+			let events = net_graph_msg_handler.get_and_clear_pending_msg_events();
+			assert_eq!(events.len(), 1);
+			match &events[0] {
+				MessageSendEvent::SendChannelRangeQuery{ node_id, msg } => {
+					assert_eq!(node_id, &node_id_1);
+					assert_eq!(msg.chain_hash, chain_hash);
+					assert_eq!(msg.first_blocknum, first_blocknum);
+					assert_eq!(msg.number_of_blocks, number_of_blocks);
+				},
+				_ => panic!("Expected MessageSendEvent::SendChannelRangeQuery")
+			};
+		}
+
+		// It should not enqueue a query when should_request_full_sync return false.
+		// The initial implementation allows syncing with the first 5 peers after
+		// which should_request_full_sync will return false
+		{
+			let (secp_ctx, net_graph_msg_handler) = create_net_graph_msg_handler();
+			let init_msg = Init { features: InitFeatures::known() };
+			for n in 1..6 {
```

Comment:
You need to iterate the for-loop until 7, otherwise it doesn't test syncing limitation to FULL_SYNCS_TO_REQUEST ?

Diff:
```
diff --git a/lightning/src/routing/network_graph.rs b/lightning/src/routing/network_graph.rs
index d6c72d08..4b0c2087 100644
--- a/lightning/src/routing/network_graph.rs
+++ b/lightning/src/routing/network_graph.rs
@@ -235,10 +235,10 @@ impl<C: Deref + Sync + Send, L: Deref + Sync + Send> RoutingMessageHandler for N
                        return ();
                }
 
-               // Check if we need to perform a full synchronization with this peer
-               if !self.should_request_full_sync(their_node_id) {
-                       return ();
-               }
+               //// Check if we need to perform a full synchronization with this peer
+               //if !self.should_request_full_sync(their_node_id) {
+               //      return ();
+               //}
 
                let first_blocknum = 0;
                let number_of_blocks = 0xffffffff;
```

Reply:
Dag-nabbit.  Fixed!  Thanks for catching that.  A minor refactor of the test bit me.